### PR TITLE
Updated LCOW custom kernel builds instructions with 4.12 support

### PR DESCRIPTION
**Pull request by soccerGB**
This change includes
1. Updated the 4.11 patch list, 9pfs patch was no longer necessary because gcs converts a hyper-v sock into fs beore using it to mount a 9pfs share.

2.  Added an additional patch for addressing PMEM support, regressed in kernel version >4.11.1
3. Added kernel build instructions for kernel 4.12

### CODE REVIEW

**Comment by rn** [comment]
Thanks, I'll give this a spin today on 4.12 and will report back

File: docs/customosbuildinstructions.md [hunk 1]
```diff
@@ -6,18 +6,29 @@ A LCOW custom Linux OS image was divided into two parts: a Linux kernel module a
...
 ## How to build custom kernel module
+Here you will find the steps to build a custom kernel for the
+Linux Hyper-V container on Windows (**LCOW**). To build the full image,
+please follow the instructions:
 
-- Have your 4.11 kernel source tree ready
+- Have your kernel source tree ready
 
-- Apply additional [4.11 patches](../kernel/README.md) to your 4.11 kernel source tree
+- Apply additional LOCW patches to your kernel source tree
```
Comment (rn) [inline docs/customosbuildinstructions.md]: LOCW -> LCOW

File: kernel/README.md [hunk 2]
```diff
@@ -1,94 +1,15 @@
-# Custom Linux kernel for LCOW
+# How to build custom Linux kernels for LCOW
 
 Here you will find the steps to build a custom kernel for the
 Linux Hyper-V container on Windows (**LCOW**). To build the full image,
 please follow the instruction from [how to produce a custom Linux OS
 image](../docs/customosbuildinstructions.md)
 
-## Patches
+## Supported Linux kerenl versions
```
Comment (rn) [inline kernel/README.md]: kerenl -> kernel

File: kernel/patches-4.11.x/README.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+this instructions is to get them from a developer repository and
+assuming you have a _Linux GIT repository_  already
+
+```
+git config --global user.name "yourname"
+git config --global user.email youremailaddress 
+ 
+git remote add -f dexuan-github https://github.com/dcui/linux.git
+ 
+git cherry-pick c248b14174e1337c1461f9b13a573ad90a136e1c
```
Comment (rn) [inline kernel/patches-4.11.x/README.md]: I would recommend adding a `-x` to all `git cherry-pick` lines as this keeps the original hash in the commit message.

File: kernel/patches-4.14.x/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# How to build 4.14.x based custom Linux kernel for LCOW
+
+You can download the Linux 4.14 source code from kernel.org
```
Comment (rn) [inline kernel/patches-4.14.x/README.md]: 4.14 is not out yet (no branch or tag) so you can't download it...

File: kernel/patches-4.12.x/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+assuming you have a _Linux GIT repository_  already
+
+```
+git config --global user.name "yourname"
+git config --global user.email youremailaddress 
+ 
+git remote add -f dexuan-github https://github.com/dcui/linux.git
+ 
+git cherry-pick 5181302de497cb7d5de37bbc84e01eca676f20d8
+git cherry-pick b54a12c4e3f18cd48314fd3851f5651446b0e6ee
```
Comment (rn) [inline kernel/patches-4.12.x/README.md]: This is already applied in 4.12.4 as `5116f5e2e05cf`. Might be worth noting.

File: docs/customosbuildinstructions.md (hunk 1 above)
Comment (soccerGB) [reply docs/customosbuildinstructions.md]: Thanks for catching this typo

File: kernel/README.md (hunk 2 above)
Comment (soccerGB) [reply kernel/README.md]: Thanks for catching this typo

File: kernel/patches-4.11.x/README.md (hunk 3 above)
Comment (soccerGB) [reply kernel/patches-4.11.x/README.md]: I didn't know that useful option. I will add that.

File: kernel/patches-4.14.x/README.md (hunk 4 above)
Comment (soccerGB) [reply kernel/patches-4.14.x/README.md]: I was basing on the most recently commit on the Linux mainline.  I will remove this 4.14 entry and wait for the source code to be available-- in the meanwhile i could get some more testing done...

**Pull request merged**